modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/portlet/action/EditRedirectPatternsMVCActionCommand.java

```diff
@@ -43,8 +49,58 @@ protected void doProcessAction(
 			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
+		try {
+			_redirectPatternConfigurationProvider.updateRedirectionPatterns(
+				ParamUtil.getLong(actionRequest, "scopePK"),
+				_getRedirectionPatterns(actionRequest));
+		}
+		catch (ConfigurationModelListenerException
+					configurationModelListenerException) {
 
+			SessionErrors.add(
+				actionRequest, configurationModelListenerException.getClass());
+
+			actionResponse.sendRedirect(
+				ParamUtil.getString(actionRequest, "redirect"));
+		}
+	}
+
+	private Map<String, String> _getRedirectionPatterns(
+		ActionRequest actionRequest) {
+
+		Map<String, String> redirectPatters = new LinkedHashMap<>();
```

Comment:
1. I thinks the variable should be called `redirectPatterns`. 
2. Is it needed the use of `LinkedHashMap` instead of `HashMap`? It is used when key sorting is required and it means lower performance and higher memory footprint.